monkey/infection_monkey/master/control_channel.py

```diff
@@ -19,37 +19,51 @@ def __init__(self, server: str, agent_id: str):
         self._control_channel_server = server
 
     def should_agent_stop(self) -> bool:
-        if not self._control_channel_server:
-            return
-
         try:
             response = requests.get(  # noqa: DUO123
                 f"{self._control_channel_server}/api/monkey_control/{self._agent_id}",
                 verify=False,
+                proxies=ControlClient.proxies,
                 timeout=SHORT_REQUEST_TIMEOUT,
             )
 
             response = json.loads(response.content.decode())
             return response["stop_agent"]
         except Exception as e:
+            # TODO: Evaluate how this exception is handled; don't just log and ignore it.
```

Comment:
When are these todo's getting fixed? If we intend to keep them, maybe it's better to open up an issue

Reply:
I'm planning to address these during the implementation of the AutomatedMaster.